golangflag.go

```diff
@@ -103,3 +103,15 @@ func (f *FlagSet) AddGoFlagSet(newSet *goflag.FlagSet) {
 	}
 	f.addedGoFlagSets = append(f.addedGoFlagSets, newSet)
 }
+
+// ParseGoTestFlags Parses go test flags (i.e. the one with '-test.') separately because pflag.Parse() skip  flags with '-test.' prefix
+// Typical usage example: `ParseGoTestFlags(os.Args[1:], goflag.CommandLine)`
+func ParseGoTestFlags(osArgs []string, goFlagSet *goflag.FlagSet) error {
+	var goTestFlags []string
+	for _, f := range osArgs {
+		if strings.HasPrefix(f, "-test.") {
```

Comment:
Is this logic intended to match up with logic we have implemented somewhere else? If so, can we make it use the same implementation for the condition (so updating it in one place won't require remembering to update it in the other)?

Reply:
Not quite, the other place is here https://github.com/spf13/pflag/blob/9edfc8d416bea8aecfbdf0790a838f95ae133078/flag.go#L1017 I guess we could externalize in a constant the `"test."` prefix?

Reply:
I think it would be valuable to extract the entire condition `strings.HasPrefix(arg, "test.")` into a helper somewhere.

Reply:
I see, there should be 2 helper conditions though: one for `"test."` and one for `"-test."` (note the leading `-` in the latter). I see the value in naming the function in a more meaningful way than "HasPrefix" but not much more than that. Anyway, I can give it a shot...